code/src/pages/MovieDetail.js

```diff
@@ -0,0 +1,64 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { NotFound } from '../components/NotFound';
+import { Loader } from '../components/Loader';
+import { BackButton } from '../components/BackButton';
+
+import '../styles/MovieDetail.css';
+
+export const MovieDetail = () => {
+  const [movie, setMovie] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const { movieId } = useParams();
+
+  const apiKey = process.env.REACT_APP_API_KEY_FOR_MOVIE;
+
+  useEffect(() => {
+    fetch(
+      `https://api.themoviedb.org/3/movie/${movieId}?api_key=${apiKey}&language=en-US`
+    )
+      .then((res) => res.json())
+      .then((json) => {
+        console.log(json);
+        setMovie(json);
+        setLoading(false);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }, [movieId, apiKey]);
+
+  if (movie.success === false) {
+    return <NotFound />;
+  }
+
+  return (
+    <>
```

Comment:
No need for React fragment over here, since you are either return `<Loader />` or `<section />`, never both of them in the same time